src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticAnalyzerExecutor.cs

```diff
@@ -215,10 +219,13 @@ private void ReportAnalyzerExceptions(Project project, ImmutableDictionary<Diagn
                 }
             }
 
-            private Dictionary<string, DiagnosticAnalyzer> CreateAnalyzerMap(IEnumerable<DiagnosticAnalyzer> analyzers)
+            private void AppendAnalyzerMap(IEnumerable<DiagnosticAnalyzer> analyzers, Dictionary<string, DiagnosticAnalyzer> analyzerMap)
             {
-                // TODO: this needs to be cached. we can have 300+ analyzers
-                return analyzers.ToDictionary(a => a.GetAnalyzerIdAndVersion().Item1, a => a);
+                foreach (var analyzer in analyzers)
+                {
+                    // user might have included exact same analyzer twice as project analyzers explicitly. we consider them as one
+                    analyzerMap[analyzer.GetAnalyzerId()] = analyzer;
```

Comment:
`GetAnalyzerIdAndVersion` was replaced with a call to `GetAnalyzerId`. Was this intentional?

Reply:
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/AnalyzerHelper.cs,132

GetAnalyzerIdAndVersion().Item1 == analyzer.GetAnalyzerId()